File: py/demo/Glacier2/callback/SessionServer.py
#!/usr/bin/env python
# **********************************************************************
#
# Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
#
# This copy of Ice is licensed to you under the terms described in the
# ICE_LICENSE file included in this distribution.
#
# **********************************************************************

import sys, traceback, Ice, Glacier2

class DummyPermissionsVerifierI(Glacier2.PermissionsVerifier):
    def checkPermissions(self, userId, password, current=None):
        print "verified user `" + userId + "' with password `" + password + "'"
        return (True, "")

class SessionI(Glacier2.Session):
    def __init__(self, userId):
        self.userId = userId

    def destroy(self, current=None):
        print "destroying session for user `" + self.userId + "'"
        current.adapter.remove(current.id)

class SessionManagerI(Glacier2.SessionManager):
    def create(self, userId, control, current=None):
        print "creating session for user `" + userId + "'"
        session = SessionI(userId)
        return Glacier2.SessionPrx.uncheckedCast(current.adapter.addWithUUID(session))

class SessionServer(Ice.Application):
    def run(self, args):
        adapter = self.communicator().createObjectAdapter("SessionServer")
        adapter.add(DummyPermissionsVerifierI(), self.communicator().stringToIdentity("verifier"))
        adapter.add(SessionManagerI(), self.communicator().stringToIdentity("sessionmanager"))
        adapter.activate()
        self.communicator().waitForShutdown()
        return 0

app = SessionServer()
sys.exit(app.main(sys.argv, "config.sessionserver"))

File: cpp/demo/Glacier2/callback/SessionServer.cpp
// **********************************************************************
//
// Copyright (c) 2003-2008 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#include <Ice/Ice.h>
#include <SessionI.h>

using namespace std;

class SessionServer : public Ice::Application
{
public:

    virtual int run(int, char*[]);
};

int
main(int argc, char* argv[])
{
    SessionServer app;
    return app.main(argc, argv, "config.sessionserver");
}

int
SessionServer::run(int argc, char* argv[])
{
    if(argc > 1)
    {
        cerr << appName() << ": too many arguments" << endl;
        return EXIT_FAILURE;
    }

    Ice::ObjectAdapterPtr adapter = communicator()->createObjectAdapter("SessionServer");
    Glacier2::PermissionsVerifierPtr dpv = new DummyPermissionsVerifierI;
    adapter->add(dpv, communicator()->stringToIdentity("verifier"));
    Glacier2::SessionManagerPtr sm = new SessionManagerI;
    adapter->add(sm, communicator()->stringToIdentity("sessionmanager"));
    adapter->activate();
    communicator()->waitForShutdown();
    return EXIT_SUCCESS;
}
